osu.Game/Online/Multiplayer/MultiplayerClient.cs

```diff
@@ -972,6 +977,20 @@ private Task runOnUpdateThreadAsync(Action action, CancellationToken cancellatio
             return tcs.Task;
         }
 
+        private void handleRoomRequest(Action request)
```

Comment:
The previous code has `forceSchedule = false` while the new flow doesn't, can one of you two confirm this is known and expected?

Re-reviewing this as a whole seems like it may take a bit of time so I'd rather just confirm this-is-okay and go with it.

Reply:
Earliest reference I can find is https://github.com/ppy/osu/pull/11316. At the time the join/leave task chain didn't exist and the scheduler was relied on as an ordering mechanism, so I don't think there's much purpose to the `forceSchedule` these days.